src/index.ts

```diff
@@ -5,6 +5,13 @@ import mongoose, { Schema, Query, Document, SchemaType } from 'mongoose';
 export interface MongooseLeanDefaultsOptions {
   defaults?: boolean;
 }
+interface Default {
+  value: unknown | undefined;
+  fnValue: ((doc: unknown) => unknown) | undefined;
+  pathSegments: string[];
+}
+
+const DEFAULTS_REGISTRY = new Map<Schema, Default[]>();
```

Comment:
I don't think it's critical since Schemas exist for the whole lifecycle of the process, but maybe just to be extra safe a `WeakMap` would be better
```suggestion
const DEFAULTS_REGISTRY = new WeakMap<Schema, Default[]>();
```

Reply:
@DouglasGabr Yeah, I was assuming that schemas are created statically. But maybe there are cases when someone is actually constructing it on the fly, let me add it!